src/main/java/com/auth0/net/client/Auth0HttpResponse.java

```diff
@@ -0,0 +1,79 @@
+package com.auth0.net.client;
+
+import com.auth0.utils.Asserts;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Auth0HttpResponse {
+
+    private final int code;
```

Comment:
Refer - https://github.com/auth0/auth0-java/pull/474#discussion_r1026619080

Reply:
This cannot be achieved since `Auth0HttpResponse` will only have a string representation of the content where as `Response` object will have parsed info or even throws exception based on received code and body. We cannot ask the custom client implementor to take care of these things. Hence `Auth0HttpResponse` will be used only to wrap and provide the required information from the custom client to the request which can parse and provide the actual response